packages/table/src/tableBody.tsx

```diff
@@ -125,53 +146,22 @@ export class TableBody extends React.Component<ITableBodyProps, {}> {
         return renderBodyContextMenu(new MenuContext(target, selectedRegions, grid.numRows, grid.numCols));
     }
 
-    private renderGhostCell = (rowIndex: number, columnIndex: number, extremaClasses: string[]) => {
-        const { grid } = this.props;
+    private renderCell = (rowIndex: number, columnIndex: number, extremaClasses: string[], isGhost: boolean) => {
+        const { cellRenderer, grid } = this.props;
+        const baseCell = isGhost ? emptyCellRenderer(rowIndex, columnIndex) : cellRenderer(rowIndex, columnIndex);
         const cell = Utils.assignClasses(
-            emptyCellRenderer(rowIndex, columnIndex),
-            TableBody.cellClassNames(rowIndex, columnIndex),
-            extremaClasses,
-            CELL_GHOST_CLASS, {
-                [CELL_LEDGER_ODD_CLASS]: (rowIndex % 2) === 1,
-                [CELL_LEDGER_EVEN_CLASS]: (rowIndex % 2) === 0,
-            });
-        const key = TableBody.cellReactKey(rowIndex, columnIndex);
-        const style = Object.assign({}, cell.props.style, Rect.style(grid.getGhostCellRect(rowIndex, columnIndex)));
-        return React.cloneElement(cell, { key, style } as ICellProps);
-    }
-
-    private renderCell = (rowIndex: number, columnIndex: number, extremaClasses: string[]) => {
-        const {
-            allowMultipleSelection,
-            grid,
-            cellRenderer,
-            onSelection,
-            selectedRegions,
-            selectedRegionTransform,
-        } = this.props;
-        const cell = Utils.assignClasses(
-            cellRenderer(rowIndex, columnIndex),
+            baseCell,
             TableBody.cellClassNames(rowIndex, columnIndex),
             extremaClasses,
             {
+                [CELL_GHOST_CLASS]: isGhost,
                 [CELL_LEDGER_ODD_CLASS]: (rowIndex % 2) === 1,
                 [CELL_LEDGER_EVEN_CLASS]: (rowIndex % 2) === 0,
             });
         const key = TableBody.cellReactKey(rowIndex, columnIndex);
-        const style = Object.assign({}, cell.props.style, Rect.style(grid.getCellRect(rowIndex, columnIndex)));
-        return (
-            <DragSelectable
-                allowMultipleSelection={allowMultipleSelection}
-                key={key}
-                locateClick={this.locateClick}
-                locateDrag={this.locateDrag}
-                onSelection={onSelection}
-                selectedRegions={selectedRegions}
-                selectedRegionTransform={selectedRegionTransform}
-            >
-                {React.cloneElement(cell, { style } as ICellProps)}
-            </DragSelectable>
-        );
+        const rect = isGhost ? grid.getGhostCellRect(rowIndex, columnIndex) : grid.getCellRect(rowIndex, columnIndex);
+        const style = Object.assign({}, cell.props.style, Rect.style(rect));
+        return React.cloneElement(cell, { style, key } as ICellProps);
```

Comment:
two clones, one object: `Utils.assignClasses` and right here. for every cell.

is it possible to merge those into one final clone, given that the first one just changes `className`?